### PR TITLE
JIT: Give `Compiler` a proper constructor

### PR DESCRIPTION
**Pull request by jakobbotsch**
- Add a proper constructor for `Compiler` and call it via placement new from `jitNativeCode`.
- Remove `compInit`, `fgInit`, `impInit`, `lvaInit` and `optInit`. Move all their initialization into the constructor or, when possible, initialize the field directly with its declaration.

### CODE REVIEW

**Comment by jakobbotsch** [comment]
cc @dotnet/jit-contrib PTAL @AndyAyersMS 

With this change `Compiler` members can be initialized directly with their declaration in `compiler.h` when the initialization value is a simple constant. More complex initialization can be done in `Compiler::Compiler` directly.

[No diffs](https://dev.azure.com/dnceng-public/public/_build/results?buildId=792193&view=ms.vss-build-web.run-extensions-tab), some minor TP diffs (the somewhat large MinOpts ones are in collections with almost no MinOpts contexts).

File: src/coreclr/jit/compiler.cpp [hunk 1]
```diff
@@ -326,6 +326,173 @@ Histogram loopExitCountTable(loopExitCountBuckets);
...
+    assert(!fJitStressRange.Error());
+    compAllowStress =
+        fJitStressRange.Contains(info.compMethodHash()) &&
+        (JitConfig.JitStressOnly().isEmpty() ||
+         JitConfig.JitStressOnly().contains(info.compMethodHnd, info.compClassHnd, &info.compMethodInfo->args));
+
+#endif // DEBUG
+
+#if defined(FEATURE_EH_WINDOWS_X86)
+    // Cache Native AOT ABI check. This must happen *after* eeInfoInitialized is initialized, above.
```
Comment (AndyAyersMS) [inline src/coreclr/jit/compiler.cpp:380]: Maybe reword this comment? `eeInfoInitialized` is now initialized inline (which I assume happens "before" the constructor body runs).

**Review by AndyAyersMS — APPROVED** [review]
Will take me a while to get used to the fact that `fgInit` is no more.

File: src/coreclr/jit/compiler.cpp (hunk 1 above)
Comment (jakobbotsch) [reply src/coreclr/jit/compiler.cpp]: Deleted the comment (indeed, all the inline initialization happens before the body here runs, so we shouldn't have concerns like this anymore)
